### PR TITLE
Create integration tests

### PR DESCRIPTION
**Pull request by krl**
Refactor the various unit tests found in semantic.rs into their own integration tests, exposing the different types as public as needed.

This is a first step towards testing more cosmwasm example contracts, and personally to get a feel for how the crate is laid out.

Draft status since this still does not add any new functionality.

### CODE REVIEW

File: vm-wasmi/src/validation.rs [hunk 1]
```diff
@@ -54,14 +53,16 @@ impl<'a> CodeValidation<'a> {
...
     /// * `input`: Custom input to validator
-    pub fn validate_module<V: Validator>(
-        self,
-        input: <V as Validator>::Input,
-    ) -> Result<Self, ValidationError> {
-        wasmi_validation::validate_module::<V>(self.0, input)
-            .map_err(ValidationError::Validation)?;
-        Ok(self)
-    }
+    // pub fn validate_module<V: Validator>(
```
Comment (dzmitry-lahoda) [inline vm-wasmi/src/validation.rs:56]: why it was commented out?

Comment (krl) [reply vm-wasmi/src/validation.rs:56]: At the moment the code used for validation is unmaintained and outdated, and does not support current wasm instruction set (especially sign_ext)

When the PR is out of draft mode i will remove all commented out code and replace with proper validation from more up-to-date libraries.

**Comment by dzmitry-lahoda** [comment]
Tests are good. I am ok with current level of CW compatiblity.

File: vm-wasmi/src/validation.rs (hunk 1 above)
Comment (dzmitry-lahoda) [reply vm-wasmi/src/validation.rs:56]: please add this as separate ISSUE and link from commented out code.

**Review by dzmitry-lahoda — APPROVED** [review]

**Pull request closed without merging**